src/cbang/os/MacOSUtilities.cpp

```diff
@@ -0,0 +1,166 @@
+/******************************************************************************\
+
+          This file is part of the C! library.  A.K.A the cbang library.
+
+                Copyright (c) 2003-2020, Cauldron Development LLC
+                   Copyright (c) 2003-2017, Stanford University
+                               All rights reserved.
+
+         The C! library is free software: you can redistribute it and/or
+        modify it under the terms of the GNU Lesser General Public License
+       as published by the Free Software Foundation, either version 2.1 of
+               the License, or (at your option) any later version.
+
+        The C! library is distributed in the hope that it will be useful,
+          but WITHOUT ANY WARRANTY; without even the implied warranty of
+        MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+                 Lesser General Public License for more details.
+
+         You should have received a copy of the GNU Lesser General Public
+                 License along with the C! library.  If not, see
+                         <http://www.gnu.org/licenses/>.
+
+        In addition, BSD licensing may be granted on a case by case basis
+        by written permission from at least one of the copyright holders.
+           You may request written permission by emailing the authors.
+
+                  For information regarding this software email:
+                                 Joseph Coffland
+                          joseph@cauldrondevelopment.com
+
+\******************************************************************************/
+
+#ifdef __APPLE__
+
+#include "MacOSUtilities.h"
+#include <CoreFoundation/CoreFoundation.h>
+//#include <errno.h>
+#include <sys/sysctl.h>
+
+// The bulk of the get macOS version code comes from a
+// stackoverflow thread http://bit.ly/3pqI8yw
+// https://gist.github.com/kbernhagen/48e89cebac618e0b2bf535e4a1a11afa
+
+static bool versionOK;
```

Comment:
My preference is to put static members in a anonymous namespace instead of using the static keyword.  The effect is the same but the namespace method is more C++y.